backend/src/apiserver/resource/resource_manager.go

```diff
@@ -386,19 +386,24 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 	}
 
 	// Check if the number of workflow CRDs is already over the specified threshold. If yes, delete the oldest workflow in the persisted final state.
-	// The cleanup of excessive workflows here is just a best-effort attempt, since we have Persistent Agent to do garbage collection.
-	workflows, err := r.getWorkflowClient(namespace).List(v1.ListOptions{})
-	if err == nil && workflows != nil && workflows.Items != nil && len(workflows.Items) >= *maximumNumberOfWorkflowCRDs {
-		oldestIndex := -1
-		for i := range workflows.Items {
-			if util.NewWorkflow(&workflows.Items[i]).PersistedFinalState() {
-				if oldestIndex < 0 || oldestIndex >= 0 && workflows.Items[i].CreationTimestamp.Time.Before(workflows.Items[oldestIndex].CreationTimestamp.Time) {
-					oldestIndex = i
+	// In addition, if flag maximumNumberOfWorkflowCRDs is 0 or negative, no enforcement of maximum amount of workflows.
+	if *maximumNumberOfWorkflowCRDs > 0 {
```

Comment:
realized another long standing problem whenever we add logic to `CreateRun`: we also need to add it to scheduled workflow controller.

Reply:
I'd personally prefer just changing default TTL in this PR, it should solve the problem for most of the cases I think.